### PR TITLE
Add option to transform JSON response

### LINKED ISSUE #24: Transform response to not use "data" as JSON
Several ES admin REST endpoints return JSON response that uses "data" as hash keys. Example of this is cluster health API that returns indices section like this:

````json
{
  "indices": {
    "index_01": { "number_of_shards": 5, "number_of_replicas": 3 },
    "index_02": { "number_of_shards": 4, "number_of_replicas": 2 }
  }
}
````

This typically occurs in all cases where the `nodes`, `indices`, `shards` or `segments` are part of the response. This can be problematic, for example, if the data are later stored into Elasticsearch then this causes several problems:

- With every new name (index, shard, node, segment, ...) the Elasticsearch mapping is modified. This can be a major issue.
- It is harder to construct useful queries and aggregations.

Introduce new watches option to modify the response data to mitigate/fix issues described above. 

---
Prior art: we were doing the same thing in old [elasticsearch-river-sysinfo](https://github.com/searchisko/elasticsearch-river-sysinfo/blob/master/tutorial/TUTORIAL.md#mappings-and-transformations).

### PR DESCRIPTION
**Pull request by lukas-vlcek**
Most commands accept `--transform=nested` option. It transforms JSON response to nested data (this means the JSON does not contain data as a JSON keys).

Make tests pass on clusters with more than one node.

Closes #24

### CODE REVIEW

**Comment by lukas-vlcek** [comment]
@richm PTAL

**Comment by lukas-vlcek** [comment]
@richm I did some modifications. The option is called `--transform=nested` now and the code is more general & prepared to support more transformation types in the future.

File: watches/commands/base.py [hunk 1]
```diff
@@ -46,6 +51,142 @@ def getResponseContentType(self):
...
 
+    def transformData(self, data):
+        """
+        Data can be transformed before sending to client.
+        Currently, the only transformation type implemented is 'nested'.
+        :param data:
+        :return:
+        """
+        transform = self.options['--transform']
+        if transform and transform is not None and transform == "nested":
```
Comment (richm) [inline watches/commands/base.py]: `if transform and transform is not None` - the part `transform is not None` is redundant - the `if transform` part already does that.
Should throw an exception or print an error or something if the value is specified and it is not equal to `"nested"`

Comment (lukas-vlcek) [inline watches/commands/base.py]: @richm I updated the code.

Unfortunately, after several attempts I gave up on implementing regular unit test for catching exceptions when unsupported `transform` option is used. _Python is still not my best friend._

**Comment by richm** [comment]
Add a test like this e.g. to tests/commands/test_cluster_state.py:
```

    def test_bogus_transform_value(self):
        cmd = self.appendSecurityCommands(['watches', 'cluster_state', '--transform=bogus'])
        output, errout = popen(cmd, stdout=PIPE, stderr=PIPE).communicate()
        self.assertRegexpMatches(errout, '(?ms)ERROR:.*RuntimeError: Unsupported transform type')
```
I think you only need to add a test like this to one of the commands that supports the `--transform` argument.

**Comment by lukas-vlcek** [comment]
Done (in `tests/commands/test_cluster_health.py`).

**Pull request merged**